### PR TITLE
fix(external-router): extract daff paths ignoring full path

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

`daffExtractDaffPathData` ignores all routes in the path except the last, incorrectly building the path when the last route is an empty path route.

## What is the new behavior?
`daffExtractDaffPathData` computes the path from all routes in the path from root.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by griest024** [comment]
@damienwebdev is this the behavior that we actually want?

File: libs/external-router/src/util/extract-daff-path-data.ts [hunk 1]
```diff
@@ -8,6 +8,6 @@ import {
  * RouteSnapshot.
  */
 export const daffExtractDaffPathData = <T extends Data = Data, K extends keyof T = keyof T>(snapshot: ActivatedRouteSnapshot, key: K): T[K] => {
-  const pathFromRoot = snapshot.pathFromRoot[snapshot.pathFromRoot.length - 1].url.map((seg) => seg.path).join('/');
+  const pathFromRoot = snapshot.pathFromRoot.flatMap((route) => route.url).map((seg) => seg.path).join('/');
```
Comment (damienwebdev) [inline libs/external-router/src/util/extract-daff-path-data.ts:11]: What was previously happening? Can you add a test for the prior state that errored? Or, does the associated test cover that state?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/external-router/src/util/extract-daff-path-data.ts (hunk 1 above)
Comment (griest024) [reply libs/external-router/src/util/extract-daff-path-data.ts:11]: the test covers that state

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**